client/post-editor/editor-term-selector/index.jsx

```diff
@@ -11,24 +11,37 @@ import { cloneDeep, findIndex, map } from 'lodash';
 import TermTreeSelector from 'my-sites/term-tree-selector';
 import AddTerm from 'my-sites/term-tree-selector/add-term';
 import { editPost } from 'state/posts/actions';
+import { resetEditorTermAdded } from 'state/ui/editor/terms/actions';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getEditorPostId } from 'state/ui/editor/selectors';
 import { getEditedPostValue } from 'state/posts/selectors';
+import { getEditorTermAdded } from 'state/ui/editor/terms/selectors';
 
 class EditorTermSelector extends Component {
 	static propTypes = {
-		siteId: React.PropTypes.number,
-		postId: React.PropTypes.number,
-		postTerms: React.PropTypes.object,
-		postType: React.PropTypes.string,
-		taxonomyName: React.PropTypes.string
+		siteId: PropTypes.number,
+		postId: PropTypes.oneOfType( [
+			PropTypes.number,
+			PropTypes.string
+		] ),
+		postTerms: PropTypes.object,
+		postType: PropTypes.string,
+		taxonomyName: PropTypes.string
 	};
 
 	constructor( props ) {
 		super( props );
 		this.boundOnTermsChange = this.onTermsChange.bind( this );
 	}
 
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.addedTerm && ( nextProps.addedTerm !== this.props.addedTerm ) ) {
+			const { siteId, postId, taxonomyName, addedTerm } = nextProps;
+			this.onTermsChange( addedTerm );
+			nextProps.resetEditorTermAdded( siteId, postId, taxonomyName );
```

Comment:
Minor: Feels a bit random that this was left out of the destructuring. Line getting a bit long I guess? Alternatively could return earn if the condition's not met to avoid the extra indentation.